experimental/FTheoryTools/src/AbstractFTheoryModels/attributes.jl

```diff
@@ -1271,21 +1271,19 @@ julia> h = euler_characteristic(qsm_model; check = false)
 378
 ```
 """
-function euler_characteristic(m::AbstractFTheoryModel; check::Bool = true)
+@attr Int ignore_kwargs=[check] function euler_characteristic(m::AbstractFTheoryModel; check::Bool = true)
```

Comment:
What exactly is the idea behind this `ignore_kwargs` list? Does it ever make sense to use that with a strict subset of accepted `kwarg`?

In all the examples I've seen here (but I did not look at all) this always contained all kwargs. Overall it seems somewhat unwieldy.

Wouldn't we perhaps be better off by changing things so that the default is that when no `ignore_kwargs` is given, all kwargs are ignored, and we amend the documentation for `@attr` to warn that if kwargs are there, they must not affect the computation, or else.

Reply:
That discussion would have better been in https://github.com/Nemocas/AbstractAlgebra.jl/pull/1958. I had hoped that you like the design since you approved that PR. 

It is mostly a precaution for people that add a kwarg and forget to think about attributes. Since in this case, it will raise an error during macro expansion. One can then either add the kwarg to the ignore_kwargs list or need to remove the attr.

I am happy to change things again, if there is a good idea and plan for that. And we need to think about the breakiness of any changes to AA (since https://github.com/Nemocas/AbstractAlgebra.jl/pull/1958 got released there)

Reply:
I support @fingolfin s suggestion to ignore keyword arguments by default and to update the documentation of `@attr`. Otherwise the code looks quite unwieldy with `@attr SomeReturnType ignorekwargs=[.......] function finally_a_function_name`. It will never fit into one line.  I find a `get_attribute!(...) do` fits well with general julia coding style. 
Another concern is that we should not introduce too many or too complicated macros. To me these kind of macros seem more like steps towards reinventing a new programming language within julia. The more macros of this kind we get, the more a programmer has to learn Oscar specifics as opposed to plain Julia.

Reply:
That being said, I greatly appreciate your work on improving the `@attr` macro. :-)